src/App.tsx

```diff
@@ -1,27 +1,92 @@
-import React from 'react';
+import React, { useCallback, useState } from 'react';
+
+import Button from 'react-bootstrap/Button';
+import Spinner from 'react-bootstrap/Spinner';
+
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { Good } from './types/Good';
+
+import { getAll, get5First, getRedGoods } from './api/goods';
+
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [hasError, setHasError] = useState(false);
+  const [lastButton, setLastButton] = useState('');
+
+  const handleGoodsLoading = useCallback(
+    (getGoods: () => Promise<Good[]>) => {
+      setHasError(false);
+      setIsLoading(true);
+
+      getGoods()
+        .then(setGoods)
+        .catch(() => setHasError(true))
+        .finally(() => setIsLoading(false));
+    },
+    [],
+  );
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+  return (
+    <div className="App">
+      <div className="header">
+        <h1 className="page-header">Dynamic list of Goods</h1>
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+        <div className="buttons-container">
+          <Button
+            type="button"
+            data-cy="all-button"
+            variant="info"
+            className="load-button"
+            onClick={() => {
+              setLastButton('all');
+              handleGoodsLoading(getAll);
+            }}
+            disabled={isLoading || lastButton === 'all'}
```

Comment:
I want to have a possibility to click on this button again if there is an error to retry